include/nbl/asset/ICommandBuffer.h

```diff
@@ -192,10 +194,13 @@ class ICommandBuffer
     }
 
     //! `_flags` takes bits from E_RESET_FLAGS
-    virtual void reset(uint32_t _flags)
+    virtual bool reset(uint32_t _flags)
     {
-        assert(m_state!=ES_PENDING);
+        if (m_state==ES_PENDING);
+            return false;
```

Comment:
that's a hard bug, not a thing you can just do and have it quietly fail and you can just handle a fail by querying the return value.

basically if you do that in Vulkan the app will crash

Reply:
@achalpandeyy change to an assert